akka-actor-typed/src/main/scala/akka/actor/typed/ActorRef.scala

```diff
@@ -20,7 +21,7 @@ import scala.util.Success
  * [[akka.event.EventStream]] on a best effort basis
  * (i.e. this delivery is not reliable).
  */
-trait ActorRef[-T] extends java.lang.Comparable[ActorRef[_]] with java.io.Serializable {
+trait ActorRef[-T] extends MessageChannel[T] with java.lang.Comparable[ActorRef[_]] with java.io.Serializable { this: InternalMessageChannel[T] ⇒
```

Comment:
Not sure if I complicated it too much with this self type for the internal stuff. Compared with just adding the internal abstract methods here. Followed similar pattern as ActorRef. I guess the advantage is that those are completely hidden for Java.